### PR TITLE
Adding State.py unit test cases

### PR DESCRIPTION
**Pull request by rahulhan**
Resubmitting state.py unit test cases.

### CODE REVIEW

**Comment by jfindlay** [comment]
@rahulhan, we're still trying to figure out why this unit test case is causing all of the other test failures that are happening.  Do you have any ideas?

**Comment by rahulhan** [comment]
@jfindlay No even i'm not sure why this happening so, but i'll review state.py unit tests once more.

**Comment by thatch45** [comment]
Go Go Jenkins!

**Comment by jfindlay** [comment]
@rahulhan, @rallytime and I are getting close to figuring out why this is causing unrelated test failures.  We think it might be caused by a problem in the test suite itself.  I'm sorry we haven't fixed it yet.  We definitely want to get your code in, but it is going to be a few days while I'm working through some other things before I can return to this, thanks.

File: tests/unit/modules/state_test.py [hunk 1]
```diff
@@ -0,0 +1,993 @@
...
+    @staticmethod
+    def rmtree(data):
+        '''
+            Mock rmtree
+        '''
+        data = data
+        return True
+
+
+state.shutil = MockShutil
```
Comment (s0undt3ch) [inline tests/unit/modules/state_test.py]: These mocks should be `@patch`ed on the test case class or `@patch`ed on `setUp` and `tearDown`.

Like this it overrides salts modules globably and other test cases need the real modules, not the patched ones.

Unittest2 does not clear `sys.modules` for each test module, that's why this is problematic.

**Comment by rahulhan** [comment]
@s0undt3ch Thanks, I'll do the required changes.

**Pull request closed without merging**